internal/controller/config/usde_config.go

```diff
@@ -0,0 +1,59 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type USDEUserStrategy string
+
+const (
+	// ProgressiveStrategyID USDEUserStrategy = "progressive" // TODO-PROGRESSIVE: enable this
+	PPNDStrategyID USDEUserStrategy = "pause-and-drain"
+)
+
+type USDEConfig struct {
+	PipelineSpecExcludedPaths   []string `json:"pipelineSpecExcludedPaths,omitempty" yaml:"pipelineSpecExcludedPaths,omitempty"`
+	ISBServiceSpecExcludedPaths []string `json:"isbServiceSpecExcludedPaths,omitempty" yaml:"isbServiceSpecExcludedPaths,omitempty"`
+}
+
+func (cm *ConfigManager) UpdateUSDEConfig(config USDEConfig) {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	cm.usdeConfig = config
+}
+
+func (cm *ConfigManager) UnsetUSDEConfig() {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	cm.usdeConfig = USDEConfig{}
+}
+
+func (cm *ConfigManager) GetUSDEConfig() USDEConfig {
+	cm.usdeConfigLock.Lock()
+	defer cm.usdeConfigLock.Unlock()
+
+	return cm.usdeConfig
+}
+
+func (s *USDEUserStrategy) UnmarshalJSON(data []byte) (err error) {
+	var usdeUserStrategyStr string
+	if err := json.Unmarshal(data, &usdeUserStrategyStr); err != nil {
+		return err
+	}
+
+	// Make sure the string is one of the possible strategy values
+	if usdeUserStrategyStr != string(PPNDStrategyID) {
+		return fmt.Errorf("invalid strategy '%s' (allowed value is: %s)", usdeUserStrategyStr, PPNDStrategyID)
+	}
+	// TODO-PROGRESSIVE: replace if-statement above for if-statement below
+	// if usdeUserStrategyStr != string(ProgressiveStrategyID) && usdeUserStrategyStr != string(PPNDStrategyID) {
+	// 	return fmt.Errorf("invalid strategy '%s' (allowed values are: %s or %s)", usdeUserStrategyStr, ProgressiveStrategyID, PPNDStrategyID)
+	// }
+
+	*s = USDEUserStrategy(usdeUserStrategyStr)
+
+	return nil
```

Comment:
nit: admittedly, I do find it a little confusing that the cluster-level configuration stuff and the user namespace-level configuration stuff are all in one file. I think this may confuse people?

Reply:
I think it would be less confusing to have all user namespace level stuff collocated

Reply:
I favored organizing by feature. In this case since the user strategy is for USDE, I put it in this file. If you think it's worth moving it, I can do that.

Reply:
no worries, I will look at possibly moving it later! thanks!